### DISCUSSION TITLE
Welcome to Vault-Job-Processor---iLogic-Extension Discussions!

### DISCUSSION THREAD

**Discussion by koechlm** [koechlm/Vault-Job-Processor---iLogic-Extension#14] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with others using or reviewing the sources provided here. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by KurbelHeld** [comment]
Hello,

very interesting add-in. Unfortunately, it doesn't seem to allow changing the material via iLogic, for example. The error message says that the material doesn't exist. Now it appears that the Inventor Server (ThisServer) is accessing other design data?! And apparently the add-in is supposed to switch the active project to the Vault project. Either this has no effect on the design data or it no longer works? In Inventor 2022 and Vault, there were also separate design data for the Inventor Server. These seem to have moved to DLLs now?

Are there any solutions for this? It would be almost essential for the results of the Inventor Server to be identical to the normal environment.

**Reply by koechlm** [reply]
Hi Kurbelheld, thank you for your feedback on this sample!
I just wanted to let you know that switching the project does not impact the design data being used per se. It depends on the setting within your Vault project file. Indeed there are significant changes in 2023, and I need to validate the impact on your objective more in detail.
Thanks,
Markus

**Reply by koechlm** [reply]
@KurbelHeld: the new location of the design data for job processor 2023, 2024, ... is C:\Program Files\Common Files\Autodesk Shared\Inventor Interoperability 2023\Design Data

**Reply by KurbelHeld** [reply]
@koechlm The design data is there, but since we use our own material libraries, it's not enough to just add them there. The project needs to be loaded for that. However, with Vault 2023, there is no more Inventor Server IPJ. Apparently, has the project now been added as fixed in the Vault Inventor Server?

**Reply by koechlm** [reply]
@KurbelHeld: I validated the Job Extension in debug mode and checked that the default project is recognized and that a Vault project file may also be used. Both worked as expected: the VaultInventorServer project returned in line: https://github.com/koechlm/Vault-Job-Processor---iLogic-Extension/blob/2023.2/Autodesk.VltInvSrv.iLogicSampleJob/iLogicJobExtension.cs#L275 is the C:\Program Files\Common Files\Autodesk Shared\Inventor Interoperability 2023\Bin\Default.ipj. The next 2 lines of code replace the default with the vaulted one. So, you can adjust the job framework behavior here according to what you need.
